test/test-onRequestFinished.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+const {deepEqual, equal, ok} = require("chai").assert;
+const sinon = require("sinon");
+
+const {setupTestDOMWindow} = require("./setup");
+
+describe("browser-polyfill", () => {
+  describe("wrapped devtools.network.onRequestFinished listener", () => {
+    it("does not wrap the listener if it is not a function", () => {
+      const fakeChrome = {
+        devtools: {
+          network: {
+            onRequestFinished: {
+              addListener: sinon.spy(),
+            },
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        const fakeNonFunctionListener = {fake: "non function listener"};
+
+        const browserOnRequestFinished = window.browser.devtools.network.onRequestFinished;
+        browserOnRequestFinished.addListener(fakeNonFunctionListener);
+
+        const fakeChromeOnRequestFinished = fakeChrome.devtools.network.onRequestFinished;
+        deepEqual(
+          fakeChromeOnRequestFinished.addListener.firstCall.args[0],
+          fakeNonFunctionListener,
+          "The non-function listener has not been wrapped"
+        );
+      });
+    });
+
+    it("promisifies the result", () => {
+      const fakeChrome = {
+        devtools: {
+          network: {
+            onRequestFinished: {
+              addListener: sinon.spy(),
+              hasListener: sinon.stub(),
+              removeListener: sinon.spy(),
+            },
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        const listener = sinon.spy();
+
+        const browserOnRequestFinished = window.browser.devtools.network.onRequestFinished;
+        browserOnRequestFinished.addListener(listener);
+
+        const fakeChromeOnRequestFinished = fakeChrome.devtools.network.onRequestFinished;
+        ok(fakeChromeOnRequestFinished.addListener.calledOnce,
+           "devtools.network.onRequestFinished.addListener should have been called once");
+
+        const wrappedListener = fakeChromeOnRequestFinished.addListener.firstCall.args[0];
+        wrappedListener({
+          getContent(cb) {
+            cb("<html>...</html>", "text/html; charset=utf8");
+          },
+        });
+
+        ok(listener.calledOnce, "listener has not been called");
+
+        const req = listener.firstCall.args[0];
+        return req.getContent().then(([content, encodingOrMimeType]) => {
```

Comment:
You should add a test to verify that the `getContent` method still accepts a callback.

Reply:
Are you suggesting that `wrappedReq.getContent(([content, encodingOrMimeType]) => { /* ... */})` should still work? Adding a test for this fails and I believe that is the way `wrapAsyncFunction()` works so I don't think we would expect callback usage to work. I also think that isn't the way Firefox does this I would think supporting that isn't necessarily expected behavior for this polyfill.

Am I misunderstanding something or are there additional changes to be made to support a callback API on top of the `Promise` API?

Reply:
Oops, never mind. We do intentionally not support callbacks in the API, per https://github.com/mozilla/webextension-polyfill#no-callback-supported-by-the-promise-based-apis-on-chrome